webpack/stories/docs/SlotAndFill.md

```diff
@@ -0,0 +1,105 @@
+# Slot And Fill
+
+`Slot & Fill` allows plugins to extend foreman core functionality in the UI
+
+## Components
+
+### Slot
+`<Slot>` is an opinionated extension point which responsible of rendering its fills
+a Slot can support multiple fills, rendering by weight order by the `multi` prop
+if there is no such prop, it will render the max weighted dfill.
+
+### Fill
+a fill is the filled object of a slot
+
+#### Component fill
+a fill that contains a child  component, which rendered by a dedicated slot
+
+*core*
+```js
+<Slot id='slot-id'>
+ a default child // can be empty
+</Slot>
+```
+
+*plugin A*
+```js
+<Fill id='slot-id' fillId='some-id' weight={100}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+*plugin B*
+```js
+<Fill id='slot-id' fillId='some-id' weight={200}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+Plugin B has a fill with a higher weight, therefore it will be rendered in a dedicated slot.
+
+#### Component fill - remdering multiple fills
+If a slot has a `multi` prop, and it has multiple fills, it will render these fills by weight order
+
+*core*
+```js
+<Slot multi id='slot-id' />
+```
+
+*plugin A*
+```js
+<Fill id='slot-id' fillId='some-id' weight={100}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+*plugin B*
+```js
+<Fill id='slot-id' fillId='some-id' weight={200}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+Plugin B's fill will be render first
+
+#### Props fill
+a fill that contains an `overrideProps` object
+those props are given to the slot's children
+
+
+*core*
+```js
+const TextWrapper = ({ text }) => <div>{text}</div>; 
+
+<Slot id='slot-id'>
+  <TextWrapper text='some default text' />
+</Slot>
+```
+
+*plugin A*
+```js
+<Fill id='slot-id' fillId='some-id' weight={200} overrideProps={{ text: '[Plugin A] this text given by a prop' }} /> 
+```
+
+
+*plugin B*
+```js
+<Fill id='slot-id' fillId='some-id' weight={100} overrideProps={{ text: '[Plugin B] this text given by a prop' }} /> 
+```
+In this case, the slot doesn't have `multi` prop, therefore it will take the max weight which is Plugin A's fill.
+
+#### Global fill
+This fill is available on each and every core page
+
+create a `fills_index.js` file under webpack directory:
```

Comment:
fills_index is not picked up by default, it needs to be imported into webpack/index.js

```js
import './fills_index';
```

Reply:
I think `fills_index` will be compiled by `webpack` into its own bundle since it ends with `*_index.js`.
Then it should be loaded into core layout in some way?
@amirfefer more information needed in this md file?

Reply:
fills_index is picked automatically by webpack - https://github.com/theforeman/foreman/pull/6554
verified  with katello

Reply:
I added this information to the md file :+1: